src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+
+import PropTypes from 'prop-types';
+import { scrollTo } from '../../helpers';
+
+import './Carousel.scss';
+export const CN = 'carousel';
+
+class Carousel extends Component {
+  constructor(props) {
+    super(props);
+    this.carouselViewport = React.createRef();
+    this.handleClick = this.handleClick.bind(this);
+    this.onResize = this.onResize.bind(this);
+    this.onScroll = this.onScroll.bind(this);
+    this.state = {
+      numOfSlidesToScroll: 3,
+      allTheWayLeft: false,
+      allTheWayRight: false,
+    };
+  }
+
+  componentDidMount() {
+    const { scrollWidth } = this.carouselViewport.current;
+    const { length } = this.props.children;
+
+    this.setState({ widthOfSlide: scrollWidth / length },
+      () => {
+        this.resizeTheCarousel();
+        this.checkNumOfSlidesToScroll();
+        this.checkIfSlidesAllTheWayOver();
+        window.addEventListener('resize', this.onResize);
+      });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.onResize);
+  }
+
+  onScroll() {
+    this.checkIfSlidesAllTheWayOver();
+  }
+
+  onResize() {
+    this.resizeTheCarousel();
+    this.checkNumOfSlidesToScroll();
+  }
+
+  resizeTheCarousel() {
+    const { widthOfSlide } = this.state;
+    let carouselWidth = '0';
```

Comment:
You can assign `${widthOfSlide * 2}px` value here and remove the last 'else' statement

Reply:
@dimas0211 just FYI